Apex7000_BillValidator/Serial/SlaveDataValidator.cs

```diff
@@ -0,0 +1,224 @@
+﻿namespace PyramidNETRS232.Serial
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    /// <summary>
+    /// Validator for slave data
+    /// </summary>
+    public class SlaveDataValidator : BaseDataValidator
+    {
+        private static readonly List<string> PossibleStates = 
+            new List<string> {"idling", "accepting", "escrowed", "stacking", "stacked", "returning", "returned"};
+        private static readonly List<string> PossibleEvents = 
+            new List<string> {"cheated", "rejected", "jammed", "stacker full", "cassette present"};
+        
+        private static readonly List<int> Credit = 
+            new List<int> {0, 1, 2, 5, 10, 20, 50, 100};
+        
+        /// <summary>
+        /// Creates a new slave data validator
+        /// </summary>
+        /// <param name="rawData"></param>
+        public SlaveDataValidator(byte[] rawData) : base(rawData)
+        {
+            
+        }
+
+        /// <summary>
+        /// List of states indicated by the state byte
+        /// </summary>
+        public List<string> States { get; protected set;} = new List<string>();
+        
+        /// <summary>
+        /// Flagged true if the acceptor is in no state.
+        /// Having a 0 for the state byte itself is not
+        /// indicative of errant data, however if the
+        /// state byte is zero, errors must be flagged as well;
+        /// i.e. Data[0] == 0 && Data[1] & 0b1110111 == 0
+        /// </summary>
+        protected bool IsNoState => !States.Any();
+
+        /// <summary>
+        /// List of events indicated by the event byte
+        /// </summary>
+        public List<string> Events { get; protected set;} = new List<string>();
+        
+        /// <summary>
+        /// List of errors indicated by the error byte
+        /// </summary>
+        public List<string> Errors { get; protected set;} = new List<string>();
+        
+        /// <summary>
+        /// Checks the state byte of the data received and populates the States
+        /// property with every state that is parsed. Adds a validation message fragment
+        /// indicating what states were parsed from the raw data.
+        /// </summary>
+        protected void CheckStates()
+        {
+            var slaveStates = new List<string>();
```

Comment:
I would just modify States directly. Clear it at the start, .Add when bits are set.